CMakeLists.txt

```diff
@@ -0,0 +1,184 @@
+	)
+
+CREATE_LIBRARY(mbedClient "${MBED_CLIENT_SRC}" "")
+	
+	
+	
+
+INCLUDE(CMakeForceCompiler)
+# CROSS COMPILER SETTING
+cmake_minimum_required (VERSION 2.8)
+SET(CMAKE_SYSTEM_NAME Generic)
+
+project(linuxPortLibs)
+
+# add the binary tree to the search path for include files
+# so that we will find TutorialConfig.h????????????????????????????????????????????????????????????????????????????
+
+set(MBED_CLIENT_C_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Port_libs/mbed-client-c)
+set(SAL_NANOSTACK_EVENT_LOOP_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Port_libs/sal-stack-nanostack-eventloop)
+set(RANDLIB_C_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Port_libs/mbed-client-randlib)
+set(NANOSTACK_LIBSERVICE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Port_libs/nanostack-libservice)
+set(NS_HAL_PAL_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Port_libs/ns-hal-pal)
+
+set(PORT_LIBS ${CMAKE_CURRENT_SOURCE_DIR})
+
+
+
+
+include_directories(${SAL_NANOSTACK_EVENT_LOOP_PATH})
+include_directories(${SAL_NANOSTACK_EVENT_LOOP_PATH}/nanostack-event-loop/)
+include_directories(${SAL_NANOSTACK_EVENT_LOOP_PATH}/nanostack-event-loop/platform/)
+include_directories(${SAL_NANOSTACK_EVENT_LOOP_PATH}/source/libNsdl/src/include/)
+include_directories(${SAL_NANOSTACK_EVENT_LOOP_PATH}/source/libCoap/src/)
+include_directories(${SAL_NANOSTACK_EVENT_LOOP_PATH}/source/libCoap/src/include/)
+
+include_directories(${MBED_CLIENT_C_PATH})
+include_directories(${MBED_CLIENT_C_PATH}/nsdl-c/)
+include_directories(${MBED_CLIENT_C_PATH}/mbed-client-c/source/libNsdl/src/)
+include_directories(${MBED_CLIENT_C_PATH}/source/libNsdl/src/include/)
+include_directories(${MBED_CLIENT_C_PATH}/source/libCoap/src/)
+include_directories(${MBED_CLIENT_C_PATH}/source/libCoap/src/include/)
+
+include_directories(${RANDLIB_C_PATH}/mbed-client-randlib)
+include_directories(${RANDLIB_C_PATH}/mbed-client-randlib/platform/)
+
+include_directories(${NANOSTACK_LIBSERVICE_PATH}/)
+include_directories(${NANOSTACK_LIBSERVICE_PATH}/mbed-client-libservice/)
+include_directories(${NANOSTACK_LIBSERVICE_PATH}/mbed-client-libservice/platform/)
+
+
+include_directories(${PORT_LIBS}/)
+include_directories(${PORT_LIBS}/mbed-trace/)
+
+include_directories(${NS_HAL_PAL_PATH})
+
+
+
+
+set(PAL_HEADER_API ${NS_HAL_PAL_PATH}/../../mbed-client-pal/Source/)
+include_directories(${PAL_HEADER_API}/PAL-Impl/Services-API/)
+#need to make this platform
+include_directories(${PAL_HEADER_API}/Port/Reference-Impl/Linux/CFStore)
+include_directories(${PAL_HEADER_API}/Port/Platform-API/)
+
+include_directories(${PAL_HEADER_API})
+
+add_definitions(-DMBED_CONF_NANOSTACK_EVENTLOOP_EXCLUDE_HIGHRES_TIMER)
+add_definitions(-DMBED_CONF_NANOSTACK_EVENTLOOP_USE_PLATFORM_TICK_TIMER)
+
+
+FILE(GLOB MBED_CLIENT_C_SRC  
+        "${MBED_CLIENT_C_PATH}/source/libNsdl/src/*.c"
+        "${MBED_CLIENT_C_PATH}/source/libNsdl/src/include/*.h"
+        
+        "${MBED_CLIENT_C_PATH}/source/libCoap/src/*.c"
+        "${MBED_CLIENT_C_PATH}/source/libCoap/src/include/*.h"
+        )
```

Comment:
We agreed that mbed-client will handle all the CMakeList dependencies for the underlying repos in this level. We are trying to avoid having the need of multiple CMakelist at different repo which may not be useful without mbed Client.